### ISSUE TITLE
gmail.observe.after 'http_event' causing hangouts and gmail to crash

### ISSUE THREAD

**Issue by jsnelgro** [KartikTalwar/gmail.js#242]
toggling this listener reliably causes the error.

**Comment by KartikTalwar** [comment]
@jsnelgro would it be possible to know how to reproduce this. Your chrome extension manifest should not be inserting gmail.js into any other domain other than mail.google.com but simply having the observer on in gmail should not be crashing it.

**Comment by jsnelgro** [comment]
@KartikTalwar this doesn't reproduce it, but is the basic gist of what I'm doing:

extension app.js
```
document.addEventListener('DOMContentLoaded', ()=> {
  var injectDOMScript = (src, cb)=> {
    script = document.createElement('script')
    script.src = chrome.extension.getURL(src)
    script.onload = script.onreadystatechange = ()=> {
      cb()
    }
    document.head.appendChild(script)
  }

  injectDOMScript('gmail.js', ()=> {
    injectDOMScript('jquery.min.js', ()=> {
      injectDOMScript('gmailjs-handler.js', ()=> {
        window.postMessage({
          from: 'my-extension'
        }, '*')
      })
    })
  })
}, false)
```

gmailjs-handler.js
```
window.addEventListener('message', (msg)=> {
  if (msg.data.from === 'my-extension') {
    var gm = window.Gmail($)
    gm.observe.after('http_event', ()=> {
      console.log('crashed!')
    })
  }
})
```

It must be something in combination with the rest of my app, but not using this listener has been a reliable fix.

**Comment by khulas18** [comment]
We are also having same issue regarding this one.
We notice that it happens when the xhr request failed.
In our project we were doing sending multiple emails at the same time on backend. During that time when our extension is installed. and we observe on http_event, the gmail will crash.
The last request on network log has a status of 503. Tried doing it multiple times to make sure if its always the case.

As a temporary fix we change this line
https://github.com/KartikTalwar/gmail.js/blob/master/src/gmail.js#L934
to
```javascript
if (this.readyState === this.DONE && progress.target.status < 400) {
```

**Comment by KartikTalwar** [comment]
hey @jsnelgro and @khulas18 could you try to make the following change and see if that fixes it for you?

for `api.tools.parse_response`

```diff
      } catch (e) {
          console.log('Gmail post response parsing failed.', e);
+          return response;
      }
```

**Comment by KartikTalwar** [comment]
I git cloned https://github.com/KartikTalwar/gmail-chrome-extension-boilerplate and updated `gmail.js` file to the latest version. Using that and pasting the following code in the console, did not crash it for me

```js
gmail.observe.after('http_event', ()=> {
  console.log('should crash')
})
```

**Comment by michi88** [comment]
I'm seeing the same behaviour and I suspect `gmail.observe.on('show_newly_arrived_message', ... `. Disabled that listener now to see if gmail keeps crashing. Also updated gmail.js from  d7f4159a22bbcc175d8db3f0fb41dbb8eadce62c to 549da3acedbc67c2b6b5132c244bf0692342f21f

**Comment by michi88** [comment]
removing `gmail.observe.on('show_newly_arrived_message'` did not help for me. What I did notice is that it happens when I click the notification of a newly arrived message and it opens in a popup (not the full browser window). You'll have to enable desktop notification as described here: https://support.google.com/mail/answer/1075549?hl=en to get the notification.

If you want to try our extension install it here: https://chrome.google.com/webstore/detail/signrequest-for-gmail/ahbbijbbhdncdlljkgjgbafcapmgdbje

Any ideas what might be causing this?

**Comment by michi88** [comment]
I narrowed it down to being a problem ONLY when a thread (or email) is opened in a popup. Try clicking the 'In new window' button next to the 'Print all' in the top right of the thread.

I'm suspecting differences in the security policies of the window but have no proof.

For now i'm disabling the plugin for popups by not inserting the scripts in the content script using:
```
if (window.opener && window.opener !== window) {
  //TODO: fix for when in popup
} else {
  // inset the needed scripts to head
}
```
What is quite interesting is that the main gmail window also crashes when a popup does. Somehow google puts the error in the session or something as it thinks some content could not be loaded. A full reload is needed.

**Comment by aesedepece** [comment]
This issue is also breaking Stampery's gmail plugin.

When clicking on a desktop notification or opening a thread in a new window, the following error #009 appears:

https://trello-attachments.s3.amazonaws.com/56f9182cac7e43b6b0ad2775/1104x92/c6587e7a57d1e9a386fba49e5ec1d00a/Screen_Shot_2016-03-29_at_10.07.14_AM.png

**Comment by aesedepece** [comment]
I believe this issue is the same as #238

**Comment by garris** [comment]
Probably fixed by #284

**Comment by josteink** [comment]
Is this still an issue for anyone? I'm closing this issue for now.

Please re-open and provide details on how to reproduce if you still have this issue.
